doc/user/content/sql/recursive-ctes.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Recursive CTEs
+description: "Recursive CTEs are used to define recursive queries."
+menu:
+  main:
+    parent: 'transform'
+    name: Recursive CTEs
+    weight: 23
+---
+
+Recursive CTEs can be used to define computations on recursively defined structures (such as trees or graphs) implied by the data in your tables.
+
+## Syntax
+
+### with_recursive_cte
+
+{{< diagram "with-recursive-ctes.svg" >}}
+
+### recursive_cte_binding
+
+{{< diagram "recursive-cte-binding.svg" >}}
+
+Field | Use
+------|-----
+**RETURN AT RECURSION LIMIT $n** | An optional clause indicating that the fixpoint computation should stop after `$n` iterations and use the current values computed for each `recursive_cte_binding` in the `select_stmt`. This could be useful when debugging and validating the correctness of recursive queries, or when you know exactly how many iterations you want to have, regardless of reaching a fixpoint. See the [Examples](#examples) section for an example.
+**ERROR AT RECURSION LIMIT $n** |  An optional clause indicating that the fixpoint computation should stop after `$n` iterations and fail the query with an error. Adding this clause with a reasonably high limit is a good safeguard against accidentally running a non-terminating dataflow in your production clusters.
+**recursive_cte_binding**  | A binding that gives the SQL fragment defined under `select_stmt` a `cte_ident` alias. This alias can be used in the same binding or in all other (preceding and subsequent) bindings in the enclosing recursive CTE block. Note that in contrast to [the `cte_binding` definition](../select#cte_binding), a `recursive_cte_binding` needs to explicitly state its type as a comma-separated list of (`col_ident` `col_type`) pairs.
+
+## Details
+
+Within a recursive CTEs block, any `cte_ident` alias can be referenced in all `recursive_cte_binding` definitions that live under the same block, as well as in the final `select_stmt` for that block.
+
+Semantically, a recursive CTE block will initialize all of its bindings to the empty collection and sequentially update them in a loop until either:
+
+1. a fixpoint is reached (i.e., if the collections stop changing across iterations), or
+1. some early exit criteria is satisfied.
```

Comment:
There are two "1."s.

Reply:
This is markdown syntax for ordered lists, they will be rendered as `<ol>` tags and the HTML engine will set them as increasing numbers. I'll change that so people reading the `*.md` file are not confused.